src/System.Runtime/tests/System/Int32Tests.cs

```diff
@@ -150,131 +150,241 @@ public static void ToString_InvalidFormat_ThrowsFormatException()
-            customFormat.CurrencySymbol = "$";
-            customFormat.NumberDecimalSeparator = ".";
-
+            
+            // Garbage strings
             yield return new object[] { null, defaultStyle, null, typeof(ArgumentNullException) };
             yield return new object[] { "", defaultStyle, null, typeof(FormatException) };
             yield return new object[] { " \t \n \r ", defaultStyle, null, typeof(FormatException) };
             yield return new object[] { "Garbage", defaultStyle, null, typeof(FormatException) };
 
-            yield return new object[] { "abc", defaultStyle, null, typeof(FormatException) }; // Hex value
-            yield return new object[] { "1E23", defaultStyle, null, typeof(FormatException) }; // Exponent
-            yield return new object[] { "(123)", defaultStyle, null, typeof(FormatException) }; // Parentheses
-            yield return new object[] { 1000.ToString("C0"), defaultStyle, null, typeof(FormatException) }; // Currency
-            yield return new object[] { 1000.ToString("N0"), defaultStyle, null, typeof(FormatException) }; // Thousands
-            yield return new object[] { 678.90.ToString("F2"), defaultStyle, null, typeof(FormatException) }; // Decimal
+            // DefaultStyle doesn't allow hex, exponents, paretheses, currency, thousands, decimal
+            yield return new object[] { "abc", defaultStyle, null, typeof(FormatException) };
+            yield return new object[] { "1E23", defaultStyle, null, typeof(FormatException) };
+            yield return new object[] { "(123)", defaultStyle, null, typeof(FormatException) };
+            yield return new object[] { 1000.ToString("C0"), defaultStyle, null, typeof(FormatException) };
+            yield return new object[] { 1000.ToString("N0"), defaultStyle, null, typeof(FormatException) };
+            yield return new object[] { 678.90.ToString("F2"), defaultStyle, null, typeof(FormatException) };
+            
+            // HexNumber
+            yield return new object[] { "0xabc", NumberStyles.HexNumber, null, typeof(FormatException) };
+            yield return new object[] { "&habc", NumberStyles.HexNumber, null, typeof(FormatException) };
+            yield return new object[] { "G1", NumberStyles.HexNumber, null, typeof(FormatException) };
+            yield return new object[] { "g1", NumberStyles.HexNumber, null, typeof(FormatException) };
+
+            // None doesn't allow hex or leading or trailing whitespace
+            yield return new object[] { "abc", NumberStyles.None, null, typeof(FormatException) };
+            yield return new object[] { "123   ", NumberStyles.None, null, typeof(FormatException) };
+            yield return new object[] { "   123", NumberStyles.None, null, typeof(FormatException) };
+            yield return new object[] { "  123  ", NumberStyles.None, null, typeof(FormatException) };
+
+            // AllowLeadingSign
```

Comment:
The comment says "AllowLeadingSign" but then some of the entries use defaultStyle and some use HexNumber... none actually use just AllowLeadingSign?